consult.el

```diff
@@ -723,35 +755,55 @@ The line beginning/ending BEG/END is bound in BODY."
         adir))))
 
 (defun consult--directory-prompt (prompt dir)
-  "Return prompt and directory.
+  "Return a cons of possibly modified PROMPT and expanded DIR.
 
 PROMPT is the prompt prefix.  The directory
 is appended to the prompt prefix.  For projects
 only the project name is shown.  The `default-directory'
 is not shown.  Other directories are abbreviated and
 only the last two path components are shown.
 
-If DIR is a string, it is returned.
-If DIR is a true value, the user is asked.
-Then the `consult-project-function' is tried.
-Otherwise the `default-directory' is returned."
-  (let* ((dir
+If DIR is null or equal to `default-directory',
+`consult--search-path-list' is set to nil.  Otherwise,
+when DIR is a string, it is split into space-separated
+path patterns, pathname wildcard expansion is performed
+on each element and a flattened list of the results is
+saved in `consult--search-path-list'.
+
+The cdr of the return value is either the expanded value
+of DIR if it is a single directory, the current project root
+directory if any from a call to `consult-project-function'
+or else `default-directory'."
+  (let* ((abbrev-dir ".")
+	 (dir
           (cond
            ((stringp dir) dir)
            (dir
             ;; Preserve this-command across `read-directory-name' call,
             ;; such that `consult-customize' continues to work.
             (let ((this-command this-command))
-              (read-directory-name "Directory: " nil nil t)))
+              ;; (read-directory-name "Directory: " nil nil t)
+	      (read-string "Search Locations (default = .): "
+			   nil nil abbrev-dir)))
```

Comment:
This change is not acceptable, since you introduce a regression for existing behavior. `read-string` is worse than `read-directory-name`. Do you have a better suggestion on how we could switch to reading multiple directories? One can use `completing-read-multiple` with a file completion table and a directory predicate, but this is generally a bit brittle.